src/biogeochem/CNPhenologyMod.F90

```diff
@@ -1776,36 +1773,30 @@ subroutine CropPhenology(num_pcropp, filter_pcropp                     , &
          ! from AgroIBIS subroutine planting
          ! ---------------------------------
 
-         ! in order to allow a crop to be planted only once each year
-         ! initialize cropplant = .false., but hold it = .true. through the end of the year
-
          ! initialize other variables that are calculated for crops
          ! on an annual basis in cropresidue subroutine
 
-         if ( jday == jdayyrstart(h) .and. mcsec == 0 )then
-
-            ! make sure variables aren't changed at beginning of the year
-            ! for a crop that is currently planted, such as
-            ! WINTER TEMPERATE CEREAL = winter (wheat + barley + rye)
-            ! represented here by the winter wheat pft
-
-            if (.not. croplive(p))  then
-               cropplant(p) = .false.
-               idop(p)      = NOT_Planted
-
-               ! keep next for continuous, annual winter temperate cereal crop;
-               ! if we removed elseif,
-               ! winter cereal grown continuously would amount to a cereal/fallow
-               ! rotation because cereal would only be planted every other year
-
-            else if (croplive(p) .and. (ivt(p) == nwwheat .or. ivt(p) == nirrig_wwheat)) then
-               cropplant(p) = .false.
-               !           else ! not possible to have croplive and ivt==cornORsoy? (slevis)
-            end if
+         ! Second condition ensures everything is correctly set when resuming from a run with old code
+         if ( is_beg_curr_year() .or. crop_inst%sdates_thisyr(p,1) == spval ) then
```

Comment:
I really appreciate your thinking about what's needed for backwards compatibility with old restart files!

Often we try to put in place backwards compatibility things like this in the `Restart` subroutine – e.g., doing something for backwards compatibility if some new variable isn't on the restart file. However, in this case, I think the way you have done it is as good or better than that approach: I'm thinking that this check is needed not just for backwards compatibility but also for some other situations: starting a run mid-year without any restart file, handling a new crop column that just came into existence (in practice that should happen at the year boundary, but it doesn't hurt to make sure that we handle that case anyway), and maybe other situations I'm not thinking of or that could arise in the future.

- [ ] If you agree that this could be handling some other edge cases like this, then can you add to the comment something like "when resuming from a run with old code, starting a run mid-year or if a new crop patch comes into existence mid-year". (That way it will be clear that we need to maintain this conditional long-term, and can't simply get rid of it once we have updated all initial conditions files.)

Reply:
Improved comment with commit 86a09e0.

Reply:
New comment looks good - thanks.